internal/controller/controlplane/kwokcontrolplane_controller.go

```diff
@@ -220,7 +220,7 @@ func (r *KwokControlPlaneReconciler) SetupWithManager(ctx context.Context, mgr c
 }
 
 func (r *KwokControlPlaneReconciler) kwokClusterToKwokControlPlane(ctx context.Context, logger *logr.Logger) handler.MapFunc {
-	return func(o client.Object) []ctrl.Request {
+	return func(ctx context.Context, o client.Object) []ctrl.Request {
```

Comment:
To fix syntax error. `handler.MapFunc` expects first argument to the function to be of type `context.Context`
```go
// MapFunc is the signature required for enqueueing requests from a generic function.
// This type is usually used with EnqueueRequestsFromMapFunc when registering an event handler.
type MapFunc func(context.Context, client.Object) []reconcile.Request
```